### PR TITLE
Decouple window from gl context

### PR DESCRIPTION
**Pull request by omni-viral**
Also remove `Surface::kind` and fix few rough points.

### CODE REVIEW

File: src/backend/gl/src/device.rs [hunk 1]
```diff
@@ -998,7 +998,20 @@ impl d::Device<B> for Device {
...
-            let _status = gl.check_framebuffer_status(target); //TODO: check status
+            
+            let status = gl.check_framebuffer_status(target);
+            match status {
+                glow::FRAMEBUFFER_COMPLETE => {},
+                glow::FRAMEBUFFER_INCOMPLETE_ATTACHMENT => panic!("One of framebuffer attachmet points are incomplete"),
+                glow::FRAMEBUFFER_INCOMPLETE_MISSING_ATTACHMENT => panic!("Framebuffer does not have any image attached"),
+                glow::FRAMEBUFFER_INCOMPLETE_DRAW_BUFFER => panic!("FRAMEBUFFER_INCOMPLETE_DRAW_BUFFER"),
+                glow::FRAMEBUFFER_INCOMPLETE_READ_BUFFER => panic!("FRAMEBUFFER_INCOMPLETE_READ_BUFFER"),
+                glow::FRAMEBUFFER_UNSUPPORTED => panic!("FRAMEBUFFER_UNSUPPORTED"),
```
Comment (kvark) [inline src/backend/gl/src/device.rs]: why can't we just `panic!("{:?}", status)` instead of repeating ourselves?

**Comment by kvark** [comment]
There is a lot in common with https://github.com/gfx-rs/gfx/pull/2882/commits/93932abf832570c239778d44f1886465ced7eff8
Looks good to me!.
bors r+

File: src/backend/gl/src/device.rs (hunk 1 above)
Comment (omni-viral) [reply src/backend/gl/src/device.rs]: Because you will get an integer

**Comment by kvark** [comment]
bors retry

**Pull request merged**